### ISSUE TITLE
[REQUEST] Allow selecting the standard library and (if libstdc++) version to use with Clang

### ISSUE THREAD

**Issue by zygoloid** [mattgodbolt/compiler-explorer#1966]
Clang supports using various different C++ standard libraries -- in particular, the corresponding version of libc++, or a wide variety of different versions of libstdc++.

The Clang installations used on compiler explorer currently default to using libstdc++ 9, but libc++ support can be requested with `-stdlib=libc++`. However, there is no way to select any other version of libstdc++ to use.

It would be great if there were some way to select a different version of libstdc++ (perhaps by adding libstdc++ to the "Libraries" list in the UI, and allowing us to pick from the same list of GCC installations that we are generally permitted to select from when picking a compiler). It'd be nice if the option to use libc++ were also exposed in the UI, but that's minor since it's already possible to get that effect via the command line.

The primary thing I'd like here is the ability to use trunk Clang with the libstdc++ from trunk GCC.

**Comment by apmorton** [comment]
Agreed that we can improve the UX on this, but in the meantime it is actually possible to do this with a little effort.

https://godbolt.org/z/HS8j_n

`--gcc-toolchain=/opt/compiler-explorer/gcc-snapshot` is the magic here

You can select whatever libstdc++ version you want by specifying the path to one of our installed compilers, which you can find:
- [by looking through our config files](https://github.com/mattgodbolt/compiler-explorer/blob/master/etc/config/c%2B%2B.amazon.properties#L14)
- or by passing `-v` to a gcc compiler on the site and observing the install path
  - note that trunk compilers are a bit special here since they are updated daily and symlinked in place
    - the latest gcc trunk is always under `/opt/compiler-explorer/gcc-snapshot`
    - the latest clang trunk is always under `/opt/compiler-explorer/clang-trunk`

**Comment by partouf** [comment]
With our new Compiler overrides interfaces, you can now change the GCC Toolchain for Clang.

![Screenshot from 2023-05-16 13-48-56](https://github.com/compiler-explorer/compiler-explorer/assets/442630/d8a36e9a-2127-46f6-9051-f41050bb8db7)
![Screenshot from 2023-05-16 13-49-25](https://github.com/compiler-explorer/compiler-explorer/assets/442630/5544321f-b686-4016-a413-8241549c68c2)
